covid_graphs/covid_graphs/country_report.py

```diff
@@ -1,24 +1,36 @@
-import numpy as np
-import os
+from dataclasses import dataclass
 from google.protobuf import text_format
+import numpy as np
+from pathlib import Path
+from typing import List
+
+from .formula import Formula
 from .pb.country_data_pb2 import CountryData
 
 
+@dataclass
+class Country:
+    name: str
+    formulas: List[Formula]
+
+
 class CountryReport:
-    """Constructs a numpy representation of data read from 'data_dir' a given country."""
-    def __init__(self, data_dir, country_tuple):
-        with open(os.path.join(data_dir, f'{country_tuple.name}.data'), "rb") as f:
-            country_data = CountryData()
-            text_format.Parse(f.read(), country_data)
-            self.name = country_data.name
-            daily_positive = np.array([day.positive for day in country_data.stats])
-            self.daily_dead = np.array([day.dead for day in country_data.stats])
-            self.daily_recovered = np.array([day.recovered for day in country_data.stats])
-            self.daily_active = daily_positive - self.daily_recovered - self.daily_dead
-            self.date_list = [
-                f"{day.date.year}-{day.date.month:02d}-{day.date.day:02d}"
-                for day in country_data.stats
-            ]
+    def __init__(self, data_dir: Path, country: Country):
+        """Constructs a numpy representation of data read from 'data_dir' a given country."""
+        country_data = CountryData()
+        data_file = Path(data_dir / f'{country.name}.data')
```

Comment:
Doesn't the file need to be closed?

Reply:
This does not open the file, just creates a path object (essentially just a smarter string). Will rename the variable for clarity.

Reply:
For completeness, from https://docs.python.org/3/library/pathlib.html:
'''
Path.read_text(encoding=None, errors=None)
...
The file is opened and then closed.
'''

Reply:
Smart!